src/frontend/src/binder/expr/mod.rs

```diff
@@ -512,6 +512,13 @@ pub fn bind_data_type(data_type: &AstDataType) -> Result<DataType> {
                 "float8" => DataType::Float64,
                 "timestamptz" => DataType::Timestamptz,
                 "jsonb" => DataType::Jsonb,
+                "serial" => {
+                    return Err(ErrorCode::NotSupported(
+                        "Column type SERIAL is not supported".into(),
+                        "Please remove the SERIAL column".into(),
```

Comment:
Maybe clearer message that user is not supposed to create a column of system type `serial`, or the `serial` type they see in table schema is not meant to be used as PostgreSQL `serial` (yet).